src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/Deflater.cs

```diff
@@ -134,9 +134,12 @@ internal int GetDeflateOutput(byte[] outputBuffer)
 
         private unsafe ZErrorCode ReadDeflateOutput(byte[] outputBuffer, ZFlushCode flushCode, out int bytesRead)
         {
+            Debug.Assert(null != outputBuffer, "Can't pass in a null output buffer!");
+            Debug.Assert(outputBuffer.Length > 0, "Can't pass in an empty output buffer!");
+        
             lock (SyncLock)
             {
-                fixed (byte* bufPtr = outputBuffer)
+                fixed (byte* bufPtr = &outputBuffer[0])
```

Comment:
Here it is just an assert - what's going to happen in Release build if the caller breaks the assertion?
NullRef? AccessViolation?